apps/web-client/src/lib/db/cr-sqlite/note.ts

```diff
@@ -292,15 +297,19 @@ export async function addVolumesToNote(db: DB, noteId: number, volume: VolumeSto
 
 	const { isbn, warehouseId, quantity } = volume;
 
-	const insertOrUpdateTxnQuery = `
-		INSERT INTO book_transaction (isbn, quantity, warehouse_id, note_id)
-		VALUES (?, ?, ?, ?)
+	const insertOrUpdateTxnStmt = `
+		INSERT INTO book_transaction (isbn, quantity, warehouse_id, note_id, updated_at)
+		VALUES (?, ?, ?, ?, ?)
 		ON CONFLICT(isbn, note_id, warehouse_id) DO UPDATE SET
 			quantity = book_transaction.quantity + excluded.quantity,
-			updated_at = (strftime('%s', 'now') * 1000)
+			updated_at = excluded.updated_at
 	`;
 
-	await db.exec(insertOrUpdateTxnQuery, [isbn, quantity, warehouseId, noteId]);
+	await db.tx(async (txDb) => {
+		const timestamp = Date.now();
+		await txDb.exec(insertOrUpdateTxnStmt, [isbn, quantity, warehouseId, noteId, timestamp]);
+		await txDb.exec(`UPDATE note SET updated_at = ? WHERE id = ?`, [timestamp, noteId]);
```

Comment:
I suggest we don't use `excluded`, but repeat the `timestamp` on line 310 to make sure `updated_at` gets a new value on update.

Reply:
I'm perfectly ok with this, but can you elaborate on the shortcomings of using `excluded.updated_at`?